crates/project-model/src/project_json.rs

```diff
@@ -88,13 +91,93 @@ pub struct Crate {
     pub(crate) exclude: Vec<AbsPathBuf>,
     pub(crate) is_proc_macro: bool,
     pub(crate) repository: Option<String>,
+    pub build: Option<Build>,
+}
+
+/// Additional, build-specific data about a crate.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct Build {
+    /// The name associated with this crate.
+    ///
+    /// This is determined by the build system that produced
+    /// the `rust-project.json` in question. For instance, if buck were used,
+    /// the label might be something like `//ide/rust/rust-analyzer:rust-analyzer`.
+    ///
+    /// Do not attempt to parse the contents of this string; it is a build system-specific
+    /// identifier similar to [`Crate::display_name`].
+    pub label: String,
+    /// Path corresponding to the build system-specific file defining the crate.
+    ///
+    /// It is roughly analogous to [`ManifestPath`], but none of the indexing/loading
+    /// operations can be directly applied/used with this file.
```

Comment:
I am not sure I can tell what indexing/loading operations are in this context

Reply:
ah, sorry: it's in reference to `ProjectManifest::from_manifest_file`. you can't run `cargo metadata` on a `rust-project.json` or Starlark-ified `TARGETS` file.

(Arguably, it should, but that's more of a discussion for https://github.com/rust-lang/rust-analyzer/pull/17246. I'll clarify this comment, sorry.)